core/arch/arm/plat-imx/pm/imx7_suspend.c

```diff
@@ -54,64 +53,15 @@ int imx7_cpu_suspend(uint32_t power_state __unused, uintptr_t entry,
 						      SUSPEND_OCRAM_OFFSET,
 						      MEM_AREA_TEE_COHERENT);
 	struct imx7_pm_info *p = (struct imx7_pm_info *)suspend_ocram_base;
-	struct suspend_save_regs *regs;
 	int ret;
 
 	if (!suspended_init) {
 		imx7_suspend_init();
 		suspended_init = 1;
 	}
 
-	regs = &reg_stack[0];
-
-	/*
-	 * Store sec ctx regs
-	 * Now we are in SVC sec mode
-	 */
-	asm volatile (
-		"mov	r2, %0\r\n"
-		"mrs	r0, cpsr\r\n"
-		"mov	r3, #0x12\r\n" /* IRQ */
-		"mov	r4, #((1 << 6) | (1 << 7))\r\n" /* I | F */
-		"orr	r3, r3, r4\r\n"
-		"msr	cpsr, r3\r\n"
-		"mrs	r1, spsr\r\n"
-		"str	r1, [r2], #4\r\n"
-		"str	sp, [r2], #4\r\n"
-		"str	lr, [r2], #4\r\n"
-		"mov	r3, #0x11\r\n" /* FIQ */
-		"orr	r3, r3, r4\r\n"
-		"msr	cpsr, r3\r\n"
-		"mrs	r1, spsr\r\n"
-		"str	r1, [r2], #4\r\n"
-		"str	sp, [r2], #4\r\n"
-		"str	lr, [r2], #4\r\n"
-		"mov	r3, #0x1b\r\n"
-		"orr	r3, r3, r4\r\n"
-		"msr	cpsr, r3\r\n"
-		"mrs	r1, spsr\r\n"
-		"str	r1, [r2], #4\r\n"
-		"str	sp, [r2], #4\r\n"
-		"str	lr, [r2], #4\r\n"
-		"mov	r3, #0x17\r\n" /* ABT */
-		"orr	r3, r3, r4\r\n"
-		"msr	cpsr, r3\r\n"
-		"mrs	r1, spsr\r\n"
-		"str	r1, [r2], #4\r\n"
-		"str	sp, [r2], #4\r\n"
-		"str	lr, [r2], #4\r\n"
-		"mov	r3, #0x16\r\n" /* MON */
-		"orr	r3, r3, r4\r\n"
-		"msr	cpsr, r3\r\n"
-		"mrs	r1, spsr\r\n"
-		"str	r1, [r2], #4\r\n"
-		"str	sp, [r2], #4\r\n"
-		"str	lr, [r2], #4\r\n"
-		"msr	cpsr, r0\r\n"	/* SVC mode */
-		:
-		: "r" (regs)
-		: "r0", "r1", "r2", "r3", "r4"
-		);
+	/* Store non-sec ctx regs */
+	sm_save_modes_regs(&mode_regs[0]);
```

Comment:
`&nsec->mode_regs` is not used at this stage and could be used instead of the local variable `mode_regs`.

Reply:
Thanks. I'll use that in next version patch after receiving more comments.